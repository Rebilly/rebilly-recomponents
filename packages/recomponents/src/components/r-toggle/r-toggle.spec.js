import {mount, shallowMount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RToggle from './r-toggle.vue';

describe('r-toggle.vue', () => {
    it('should render Wrapper and match snapshot', async () => {
        const wrapper = shallowMount(RToggle, {
            propsData: {
                name: 'snapshot',
                label: 'Custom label',
                id: 'id1234',
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render via SSR and match snapshot', async () => {
        const wrapper = renderToString(RToggle, {
            propsData: {
                name: 'snapshot',
                label: 'Custom label',
                id: 'id1234',
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('renders label when passed', () => {
        const wrapper = shallowMount(RToggle, {
            propsData: {label: 'custom label'},
        });
        expect(wrapper.text()).toMatch('custom label');
    });

    it('should render non-toggled component', () => {
        const wrapper = mount(RToggle, {
            propsData: {
                checked: null,
            },
        });

        wrapper.find('.r-toggle-switch').trigger('change');
        expect(wrapper.attributes('aria-checked')).toBeFalsy();
    });

    it('should handle change event for array', () => {
        const wrapper = mount(RToggle, {
            propsData: {
                name: 'toggle',
                label: 'Label',
                checked: [false, true],
            },
        });

        wrapper.find('.r-toggle-switch').trigger('change');
        expect(wrapper.emitted('change')).toBeTruthy();
    });

    it('should handle change event for single', () => {
        const wrapper = mount(RToggle, {
            propsData: {
                name: 'toggle',
                label: 'Label',
                checked: true,
            },
        });

        wrapper.find('.r-toggle-switch').trigger('change');
        expect(wrapper.emitted('change')).toBeTruthy();
    });
});
