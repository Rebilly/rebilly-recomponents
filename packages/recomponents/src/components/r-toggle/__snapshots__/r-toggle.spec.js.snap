// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-toggle.vue should render Wrapper and match snapshot 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        aria-checked="false"
        class="r-toggle"
        id="id1234"
        role="checkbox"
      >
        <label
          class="r-toggle-body"
          for="snapshot"
        >
          <input
            class="r-toggle-switch"
            id="snapshot"
            type="checkbox"
          />
          <span
            class="r-toggle-switch-handle"
          />
        </label>
        <label
          class="r-toggle-label"
          for="snapshot"
        >
          Custom label
        </label>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
          "props": undefined,
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": undefined,
          "render": [Function],
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.1.1",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "change": [Function],
    "checked": false,
    "disabled": false,
    "getModelValue": [Function],
    "hasOwnProperty": [Function],
    "isChecked": false,
    "label": "Custom label",
    "name": "snapshot",
    "value": "on",
  },
  "rootVM": Object {},
  "wrapperElement": <div
    aria-checked="false"
    class="r-toggle"
    id="id1234"
    role="checkbox"
  >
    <label
      class="r-toggle-body"
      for="snapshot"
    >
      <input
        class="r-toggle-switch"
        id="snapshot"
        type="checkbox"
      />
      <span
        class="r-toggle-switch-handle"
      />
    </label>
    <label
      class="r-toggle-label"
      for="snapshot"
    >
      Custom label
    </label>
  </div>,
}
`;

exports[`r-toggle.vue should render via SSR and match snapshot 1`] = `<div role="checkbox" id="id1234" class="r-toggle"><label for="snapshot" class="r-toggle-body"><input type="checkbox" id="snapshot" value="on" class="r-toggle-switch"> <span class="r-toggle-switch-handle"></span></label> <label for="snapshot" class="r-toggle-label">Custom label</label></div>`;
